### ISSUE TITLE
[Play Integrity API] Binding to the service in the play store failed . This is due to having an old play store version installed on the device

### ISSUE THREAD

**Issue by amlan007** [microg/GmsCore#2050]
Binding to the service in the play store failed . This is due to having an old play store version 

Is this fake store issue or gms core issue

**Comment by mar-v-in** [comment]
I transferred this issue to GmsCore as the FakeStore source code is not maintained in the vending-app folder in GmsCore repository.

**Comment by kuba-orlik** [comment]
For context and searchability, here are logs that appear in logcat if you try the [hydra](https://github.com/linuxct/hydra) app with lineage-microg:

```
PlayIntegrityService: Requesting Play Integrity Standard (express) token
PlayCore: UID: [10317]  PID: [23304] StandardIntegrity : warmUpIntegrityToken(<redacted>)
PlayCore: UID: [10317]  PID: [23304] StandardIntegrity : Already connected to the service.
PlayCore: UID: [10317]  PID: [23304] StandardIntegrity : Initiate binding to the service.
ActivityManager: Unable to start service Intent { act=com.google.android.play.core.expressintegrityservice.BIND_EXPRESS_INTEGRITY_SERVICE pkg=com.android.vending } U=0: not found
PlayCore: UID: [10317]  PID: [23304] StandardIntegrity : Failed to bind to the service.
CoreBackPreview: Window{3ee5e3a u0 Toast}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@b542406, mPriority=0, mIsAnimationCallback=false}
gralloc4: Unable to set buffer name VRI[Toast]#7385(BLAST Consumer)7385: File name too long
Choreographer: Frame time is 0.026415 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
gralloc4: Unable to set buffer name VRI[Toast]#7385(BLAST Consumer)7385: File name too long
PlayIntegrityService: Play Integrity API failure
PlayIntegrityService: -9: Standard Integrity API error (-9): Binding to the service in the Play Store has failed. This can be due to having an old Play Store version installed on the device.
PlayIntegrityService: Ask the user to update Play Store.
PlayIntegrityService:  (https://developer.android.com/google/play/integrity/reference/com/google/android/play/core/integrity/model/StandardIntegrityErrorCode.html#CANNOT_BIND_TO_SERVICE).
PlayIntegrityService: --------------------
```

**Comment by TailsFanLOL** [comment]
I see that #2611 is closed, is this done or nah?

**Comment by 320kg** [comment]
I read some of #2599 and it looks like it is going the same way the official implementation goes. 

My understanding:
1. Communicate back and forth with device info and eventually get some bytecode and an APK, which is a wrapper around a native library that containes the vm for the bytecode (encoding changes every week and is compiled in)
2. Run a method in the APK which passes it to the native lib, which executes the bytecode, which in itself puts the string "Come work for us droidguard.uniqueID@google.com" into memory then does some more witchery fuckery and it returns a token

Either way it means that yall are running arbitrary, proprietary code from Google's servers. Shouldn't the user get warned in any way?

**Comment by paolo-caroni** [comment]
It seems to be done on the next release (3.6).
Droidguard was always proprietary, and the users are warned. I suppose that in the next release will be cleared that.

**Comment by mar-v-in** [comment]
@TailsFanLOL this feature is disabled by default. On the screen to enable it (which currently is the same screen we use for SafetyNet) it is explained that the proprietary droidguard is used. We might want to make this more explicit and adjust the wording for Play Integrity, but in the end, if users need it, it's for a proprietary app, so users already do run proprietary code on their device anyway.

From what we've seen in the past, the droidguard proprietary code is not as much of a privacy issue, compared to regular tracking libraries you see in almost every proprietary app these days. There's been a bunch of research and analysis around droidguard, so you can probably find some good sources on that.

**Comment by TailsFanLOL** [comment]
@mar-v-in, aight. I found out how it works from a 2019 habr post, no idea if anything changed.

**Comment by ale5000-git** [comment]
This is already fixed in the version 0.3.6 of microG.

**Comment by ale5000-git** [comment]
This issue was only for "Binding to the service in the play store failed" that got implemented in v0.3.6 of microG so it is now fixed.
If there are futher issues I suggest to write on #2700 (if it is related to it) or alternatively open a new issue.
